pkg/aws/sts.go

```diff
@@ -79,7 +80,7 @@ func (c *awsClient) ValidateRoleARNAccountIDMatchCallerAccountID(roleARN string)
 }
 
 func (c *awsClient) HasPermissionsBoundary(roleName string) (bool, error) {
-	output, err := c.iamClient.GetRole(&iam.GetRoleInput{
+	output, err := c.iamClient.GetRole(context.Background(), &iam.GetRoleInput{
```

Comment:
@den-rgb Personally, I think we should go with `context.TODO()` rather than `context.Background()` as it feels most appropriate given the context API docs:

```
TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).
```
https://pkg.go.dev/context#TODO

Reply:
Given the nature of IAM role management and that the operations are routine without specific timing requirements, using context.Background() seems appropriate in this context

Reply:
although I doubt it'll make much of a difference whichever is used

Reply:
+1 I agree.